libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -152,6 +152,27 @@ bool IRGeneratorForStatements::visit(TupleExpression const& _tuple)
 	return false;
 }
 
+bool IRGeneratorForStatements::visit(IfStatement const& _ifStatement)
+{
+	_ifStatement.condition().accept(*this);
+	string condition = expressionAsType(_ifStatement.condition(), *TypeProvider::boolean());
+
+	if (_ifStatement.falseStatement())
+		m_code << "switch " << condition << "\n" << "case 1 {\n";
+	else
+		m_code << "if " << condition << " {\n";
+	_ifStatement.trueStatement().accept(*this);
+	m_code << "}\n";
+
+	if (_ifStatement.falseStatement())
+	{
+		m_code << "case 0 {\n";
+		_ifStatement.falseStatement()->accept(*this);
+		m_code << "}\n";
+	}
+	return false;
```

Comment:
Do we want to be paranoid and revert on `default`?

Reply:
Hm... that probably can't be optimized away later, can it? We could use ``case 0`` and ``default`` instead... I'm not sure what's better. I was in fact also wondering whether it's better to use a ``switch`` or one ``if cond`` and one ``if iszero(cond)``, but I think it's easier to convert the switch to something else in the optimizer than the other way round...

Reply:
I think `case 0` and `default` would be best.